File: AdventOfCode/Day4_tests.py
import unittest

import Day4

class Day4_tests(unittest.TestCase):
    def test_case_1(self):
        secret_key = "abcdef"
        expected = 609043
        actual = Day4.lowest_salt_with_leading_zeroes(secret_key)
        self.assertEqual(expected, actual)

    def test_case_2(self):
        secret_key = "pqrstuv"
        expected = 1048970
        actual = Day4.lowest_salt_with_leading_zeroes(secret_key)
        self.assertEqual(expected, actual)

if __name__ == '__main__':
    unittest.main(verbosity=2)

File: AdventOfCodeLibTests/Day4Tests.cpp
#include "stdafx.h"
#include "CppUnitTest.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

#include "Day4.h"

namespace AdventOfCodeLibTests
{		
	TEST_CLASS(Day4Tests)
	{
	public:
		
		TEST_METHOD(day4_test_case_1)
		{
            auto secret_key = "abcdef";
            auto expected = 609043;
            auto actual = Day4::lowest_salt_with_leading_zeroes(secret_key);
            Assert::AreEqual(expected, actual);
		}

        TEST_METHOD(day4_test_case_2)
        {
            auto secret_key = "pqrstuv";
            auto expected = 1048970;
            auto actual = Day4::lowest_salt_with_leading_zeroes(secret_key);
            Assert::AreEqual(expected, actual);
        }

	};
}